src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcTransport.cs

```diff
@@ -69,6 +70,13 @@ public static Stream Connect(IpcEndpointConfig config, TimeSpan timeout)
                 socket.Connect(new IpcUnixDomainSocketEndPoint(config.Address), timeout);
                 return new ExposedSocketNetworkStream(socket, ownsSocket: true);
             }
+            else if (config.Transport == IpcEndpointConfig.TransportType.TcpClient)
+            {
+                var addressPort = config.Address.Split(':');
```

Comment:
You could construct a [`System.Uri`](https://docs.microsoft.com/en-us/dotnet/api/system.uri?view=net-6.0) instead and it will handle splitting things out for you. Then you don't need to worry about someone entering `tcp://localhost:5051` or something.

Reply:
I leveraged the existing `IpcTcpSocketEndPoint` constructor to handle the address per @lateralusX's advice.